plutus-tx-plugin/test/Blueprint/Acme.golden.json

```diff
@@ -0,0 +1,119 @@
+{
+  "$schema": "https://cips.cardano.org/cips/cip57/schemas/plutus-blueprint.json",
+  "$vocabulary": {
+    "https://cips.cardano.org/cips/cip57": true,
+    "https://json-schema.org/draft/2020-12/vocab/applicator": true,
+    "https://json-schema.org/draft/2020-12/vocab/core": true,
+    "https://json-schema.org/draft/2020-12/vocab/validation": true
+  },
+  "preamble": {
+    "title": "Acme Contract",
+    "description": "A contract that does something awesome",
+    "version": "1.1.0",
+    "plutusVersion": "v3",
+    "license": "MIT"
+  },
+  "validators": [
+    {
+      "title": "Acme Validator",
+      "description": "A validator that does something awesome",
+      "redeemer": {
+        "title": "Acme Redeemer",
+        "description": "A redeemer that does something awesome",
+        "purpose": {
+          "oneOf": [
+            "spend",
+            "mint"
+          ]
+        },
+        "schema": {
+          "$ref": "#/definitions/BuiltinString"
+        }
+      },
+      "datum": {
+        "title": "Acme Datum",
+        "description": "A datum that contains something awesome",
+        "purpose": "spend",
+        "schema": {
+          "$ref": "#/definitions/Datum"
+        }
+      },
+      "parameters": [
+        {
+          "title": "Acme Parameter",
+          "description": "A parameter that does something awesome",
+          "purpose": "spend",
+          "schema": {
+            "$ref": "#/definitions/Params"
+          }
+        }
+      ],
+      "compiledCode": "01010032222323232300349103505435003232325333573466e1d200000218000a999ab9a3370e90010010c00cc8c8c94ccd5cd19b874800000860026eb4d5d0800cdd71aba13574400213008491035054310035573c0046aae74004dd51aba100109802a481035054310035573c0046aae74004dd50029919192999ab9a3370e90000010c0004c0112401035054310035573c0046aae74004dd5001119319ab9c00180019999a891001111400400e00500110948014004900126101030048810001",
+      "hash": "82c3f1e456b3856514af88ec975abe5e105d673ca51bf36101d7bd76"
+    }
+  ],
+  "definitions": {
+    "BuiltinByteString": {
+      "dataType": "bytes"
+    },
+    "BuiltinData": {},
+    "BuiltinString": {
+      "dataType": "#string"
+    },
+    "Bytes": {
+      "title": "SchemaBytes",
+      "dataType": "bytes"
+    },
+    "Datum": {
+      "oneOf": [
+        {
+          "$comment": "DatumLeft",
+          "constructor": 0,
```

Comment:
Good catch and good point! I don't think that we need to test the golden schema every CI build, it should suffice to test it once [a change];

Reply:
Why not? it should be fast and easy, right? We just need to write a little check that uses a schema checker from nixpkgs to check the schema against this output, and add it to our nix checks. I think that's probably the best way since there isn't a convenient pure-Haskell schema checker as far as I can tell :thinking:

Reply:
We can, and I like that such a schema check could be shared, as it turns out that doing schema validation isn't that straightforward, this way I can set it up once and then anybody who's changing the golden file can use it. OTOH, golden file is our expectation, its a form of a spec. It feels a bit wasteful to test if our expectation is actually a valid one for every PR. But maybe its me thinking of a premature optimisation.

Reply:
- If it isn't checked, it's broken
- Checking it periodically is harder than checking it always

:shrug:

Reply:
You convinced me, will do.